zebra-state/src/service.rs

```diff
@@ -229,6 +229,31 @@ impl StateService {
             relevant_chain,
         )?;
 
```

Comment:
TODO: move this code to the `service::check` module, in a new function that gets passed the prepared block and finalized state

Reply:
Oh no I want to move the code, I just ran out of time.

(And I wanted an initial review before I put more work in.)

Reply:
Moved in ef1d2fb4

Reply:
Oh, sorry about that. I thought it was a question :joy: